pkg/controller/appstatus/endpoints.go

```diff
@@ -54,17 +54,29 @@ func serviceEndpoints(ctx context.Context, c kclient.Client, app *v1.AppInstance
 					portNum := port.Port
 					if len(ingress.Ports) > 0 {
 						for _, ingressPort := range ingress.Ports {
-							portProtocol := port.Protocol
+							portProtocol := protocol
 							if ingressPort.Protocol != "" {
-								portProtocol = ingressPort.Protocol
+								switch ingressPort.Protocol {
```

Comment:
The protocol needs to be converted to acornv1.protocol as it is uppercase before.